packages/esm-ward-app/src/ward-workspace/patient-transfer-bed-swap/patient-transfer-request-form.component.tsx

```diff
@@ -0,0 +1,239 @@
+import { createEncounter } from '../../ward.resource';
+import useWardLocation from '../../hooks/useWardLocation';
+import type { ObsPayload } from '../../types';
+import { useInpatientRequest } from '../../hooks/useInpatientRequest';
+import { Form, ButtonSet, Button, TextArea, InlineNotification, RadioButtonGroup, RadioButton } from '@carbon/react';
+
+export default function PatientTransferForm({
+  closeWorkspaceWithSavedChanges,
+  patient,
+  patientUuid,
+  promptBeforeClosing,
+}: WardPatientWorkspaceProps) {
+  const { t } = useTranslation();
+  const [showErrorNotifications, setShowErrorNotifications] = useState(false);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const { emrConfiguration, isLoadingEmrConfiguration, errorFetchingEmrConfiguration } = useEmrConfiguration();
+  const { currentProvider } = useSession();
+  const { location } = useWardLocation();
+  const dispositionsWithTypeTransfer = useMemo(
+    () => emrConfiguration?.dispositions.filter(({ type }) => type === 'TRANSFER'),
+    [emrConfiguration],
+  );
+  const { mutate: mutateAdmissionLocation } = useAdmissionLocation();
+  const { mutate: mutateInpatientRequest } = useInpatientRequest();
+
+  const zodSchema = useMemo(
+    () =>
+      z.object({
+        location: z.string({
+          required_error: t('pleaseSelectTransferLocation', 'Please select transfer location'),
+        }),
+        note: z.string().optional(),
+        transferType:
+          dispositionsWithTypeTransfer?.length > 1
+            ? z.string({
+                required_error: t('pleaseSelectTransferType', 'Please select transfer type'),
+              })
+            : z.string().optional(),
+      }),
+    [t, dispositionsWithTypeTransfer],
+  );
+
+  type FormValues = z.infer<typeof zodSchema>;
+
+  const formDefaultValues: Partial<FormValues> = useMemo(() => {
+    const defaultValues: FormValues = {};
+    if (dispositionsWithTypeTransfer?.length === 1) {
+      defaultValues.transferType = dispositionsWithTypeTransfer[0].uuid;
+    }
+    return defaultValues;
+  }, [dispositionsWithTypeTransfer]);
+
+  const {
+    formState: { errors, isDirty },
+    control,
+    handleSubmit,
+    getValues,
+    setValue,
+  } = useForm<FormValues>({ resolver: zodResolver(zodSchema), defaultValues: formDefaultValues });
+
+  useEffect(() => {
+    if (dispositionsWithTypeTransfer?.length === 1) {
+      setValue('transferType', dispositionsWithTypeTransfer[0].uuid);
+    }
+  }, [dispositionsWithTypeTransfer]);
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+    return () => promptBeforeClosing(null);
+  }, [isDirty]);
+
+  const onSubmit = useCallback(
+    (values: FormValues) => {
+      setIsSubmitting(true);
+      setShowErrorNotifications(false);
+      const obs: Array<ObsPayload> = [
+        {
+          concept: emrConfiguration.dispositionDescriptor.internalTransferLocationConcept.uuid,
+          value: values.location,
+        },
```

Comment:
(sorry for the slew of comments, deleted the other ones, this is the latest:)

We will make some changes @vasharma05  to support the above. If you are looking to merge before then, can you comment out the above concept/value pair until we add proper support for locations (hopefully early next week)?  Thanks!

Reply:
Hi @mogoodrich, since the locations will not be present, the user will not be able to save the form, I believe it should be alright with the above condition.
Thanks!